source/BuildingBlocks.Domain/Models/ChargeType.cs

```diff
@@ -13,39 +13,19 @@
 // limitations under the License.
 
 using System;
-using System.Linq;
+using Energinet.DataHub.EDI.BuildingBlocks.Domain.DataHub;
 
 namespace Energinet.DataHub.EDI.BuildingBlocks.Domain.Models;
 
 [Serializable]
-public class ChargeType : EnumerationType
+public class ChargeType : EnumerationTypeWithCode<ChargeType>
```

Comment:
```suggestion
public class ChargeType : DataHubType<ChargeType>
```

Please let me try to explain this concept tomorrow :)